ledger/acctupdates.go

```diff
@@ -246,66 +255,36 @@ func (au *accountUpdates) waitAccountsWriting() {
 	au.accountsWriting.Wait()
 }
 
+// close closes the accountUpdates, waiting for all the child go-routine to complete
 func (au *accountUpdates) close() {
 	if au.ctxCancel != nil {
 		au.ctxCancel()
 	}
 	au.waitAccountsWriting()
+	// this would block until the commitSyncerClosed channel get closed.
+	<-au.commitSyncerClosed
```

Comment:
Should this be inside a switch?
In case this statement is waiting before 	

```
	// initialize the commitSyncerClosed with a closed channel ( since the commitSyncer go-routine is not active )
	au.commitSyncerClosed = make(chan struct{})
	close(au.commitSyncerClosed)
```

Reply:
> Should this be inside a switch?
You mean "inside a select", right ? Then the answer is "no". The `loadFromDisk` and `close` are expected to be synchronized by the caller, so there is no need to worry for thread safety between them.

Reply:
Okay. It is also synchronized with `initialize` right?

Reply:
So, `close` will run in parallel **only** with `commitSyncer` ?

Reply:
that's correct.

Reply:
The `initialize` is internal function called by `loadFromDisk`